### PR TITLE
Clean up changes in #40914

### PR DESCRIPTION
**Pull request by layomia**
Fixup for https://github.com/dotnet/runtime/pull/40914. More detail in https://github.com/dotnet/runtime/pull/41366#issuecomment-682160958.

- Prevents test issues due to non-deterministic reflection order on serialization.
- Prevents a ~6% throughput regression when using custom converters on deserialization.

### CODE REVIEW

**Comment by devsko** [comment]
Sorry, didn't know `JsonTestHelper`.

**Comment by layomia** [comment]
> Sorry, didn't know `JsonTestHelper`.

No worries at all!

File: src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyInfo.cs [hunk 1]
```diff
@@ -398,5 +399,13 @@ public JsonClassInfo RuntimeClassInfo
...
 
         public JsonNumberHandling? NumberHandling { get; private set; }
+
+        /// <summary>
+        /// Returns true if the property's converter is external (a user's custom converter)
+        /// and the type to convert is not the same as the declared property type (polymorphic).
+        /// Used to determine whether to perform additional validation on the value returned by the
+        /// converter on deserialization. A field, not property is used for a faster runtime check in a hot path.
+        /// </summary>
+        protected bool ConverterIsExternalAndPolymorphic;
```
Comment (ericstj) [inline src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyInfo.cs:409]: Could be private.

**Review by ericstj — APPROVED** [review]

**Review by ericstj — APPROVED** [review]

**Review by jozkee — APPROVED** [review]
Lgtm.

**Comment by layomia** [comment]
Test failure: https://github.com/dotnet/runtime/issues/41485.

**Pull request merged**